### PR TITLE
W4 - nhatnguyen - Create ViewPager + TabLayout 

### PR DESCRIPTION
**Pull request by at-nhatnguyen**
#### Related issues (Optional)
Task #54 

#### Description (Required)
- [x] Create intro screen with viewpager
- [x] Create screen activity with viewpager and tablayout

### CODE REVIEW

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewPagerMainActivity.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package vn.asiantech.internship.ui.viewpager;
+
+import android.app.FragmentManager;
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewPagerMainActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.app.FragmentManager'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/VocabularyAdapter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package vn.asiantech.internship.ui.viewpager.fragment;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentStatePagerAdapter;
+import android.view.View;
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/VocabularyAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.view.View'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/VocabularyAdapter.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import android.view.ViewGroup;
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/VocabularyAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.view.ViewGroup'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-nhatnguyen/res/layout/activity_viewpager.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        android:id="@+id/pageIndicatorView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignParentBottom="true"
+        android:layout_marginBottom="10dp"
+        attrs:piv_animationType="swap"
+        attrs:piv_interactiveAnimation="true"
+        attrs:piv_viewPager="@+id/viewPagerHome" />
+
+    <TextView
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/res/layout/activity_viewpager.xml]: WARNING: [RelativeOverlap] `@id/tvSkip` can overlap `@id/pageIndicatorView` if @string/skip grows due to localized text expansion
 If relative layout has text or button items aligned to left and right sides they can overlap each other due to localized text expansion unless they have mutual constraints like `toEndOf`/`toStartOf`.

**Comment by at-nhatnguyen** [comment]
@at-quocnguyen please review my task, thank you

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/MainActivity.java [hunk 5]
```diff
@@ -8,20 +8,20 @@
 
 import vn.asiantech.internship.R;
 import vn.asiantech.internship.ui.calculator.CalculatorActivity;
-import vn.asiantech.internship.ui.imageview.ImageActivity;
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/MainActivity.java]: WARNING: [DuplicateImports] Avoid duplicate imports such as 'vn.asiantech.internship.ui.login.LoginActivity'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#DuplicateImports

**Comment by at-nhatnguyen** [comment]
@at-quocnguyen please review my task, thank you

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/models/Vocabulary.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package vn.asiantech.internship.models;
+
+/**
+ * Created by hoangnhat on 14/12/2017.
+ * Class use for viewpager of homeFragment
+ */
+public class Vocabulary {
+    private String english;
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/models/Vocabulary.java]: So... Please when you declare name of model should be clearly like `private String language`

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        Dialog dialog = super.onCreateDialog(savedInstanceState);
+        dialog.setContentView(R.layout.update_dialog);
+        float dialogWidth = (float) (getActivity().getWindowManager().getDefaultDisplay().getWidth() * 0.9);
+        float dialogHeight = (float) (getActivity().getWindowManager().getDefaultDisplay().getHeight() * 0.9);
+        if (dialog.getWindow() != null) {
+            dialog.getWindow().setLayout((int) dialogWidth, (int) dialogHeight);
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java]: The best way implement in here please follow below 
```Java
@NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Dialog dialog = super.onCreateDialog(savedInstanceState);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setContentView(R.layout.fragment_edit_info_dialog);
        final Window window = dialog.getWindow();
        if (window != null) {
            dialog.getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            dialog.getWindow().setLayout(((int) (ScreenUtil.getWidthScreen(getActivity()) * 0.8)),
                    ((int) (ScreenUtil.getHeightScreen(getActivity()) * 0.8)));
        }
        dialog.setCanceledOnTouchOutside(true);
        return dialog;
    }
```

Comment (at-quocnguyen) [reply app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java]:
```Java
public final class ScreenUtil {
    private ScreenUtil() {
        // No-opp
    }

    /**
     * This method is used to get height of screen
     *
     * @param context is current context
     * @return return height screen in pixel
     */
    public static int getHeightScreen(Context context) {
        WindowManager wm = (WindowManager) context
                .getSystemService(Context.WINDOW_SERVICE);
        DisplayMetrics dimension = new DisplayMetrics();
        wm.getDefaultDisplay().getMetrics(dimension);
        return dimension.heightPixels;
    }

    /**
     * This method is used to get width of screen
     *
     * @param context is current context
     * @return return width of screen in pixel
     */
    public static int getWidthScreen(Context context) {
        WindowManager wm = (WindowManager) context
                .getSystemService(Context.WINDOW_SERVICE);
        DisplayMetrics dimension = new DisplayMetrics();
        wm.getDefaultDisplay().getMetrics(dimension);
        return dimension.widthPixels;
    }
}
```

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewpagerActivity.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * Activity use for intro
+ */
+public class ViewpagerActivity extends AppCompatActivity {
+    private List<Screen> mScreens;
+    private ScreenPagerAdapter mScreenPagerAdapter;
+    private ViewPager mViewPager;
+    private int mPositionPager;
+    private boolean lastPager;
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewpagerActivity.java]: If the variable is type boolean pls make sure have prefix `mIs...`

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewpagerActivity.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+    private void initAdapter() {
+        mScreenPagerAdapter = new ScreenPagerAdapter(mScreens, this);
+        mViewPager.setAdapter(mScreenPagerAdapter);
+        PageIndicatorView pageIndicatorView = findViewById(R.id.pageIndicatorView);
+        pageIndicatorView.setAnimationType(AnimationType.DROP);
+        pageIndicatorView.setViewPager(mViewPager);
+    }
+
+    private void listenViewPager() {
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/ViewpagerActivity.java]: better way pls follow below for check last item 
```Java
 private void checkedLastPageSwitchHomeScreen() {
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {

            int mLastPos;
            boolean isFinish = false;

            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
                isFinish = mLastPos == mHowToUsedAdapter.getCount() - 1;
                mLastPos = position;
            }

            @Override
            public void onPageSelected(int position) {

            }

            @Override
            public void onPageScrollStateChanged(int state) {
                if (isFinish) {
                    Intent intent = new Intent(TutorialActivity.this, MainActivity.class);
                    startActivity(intent);
                    isFinish = false;
                }
            }
        });
    }
```

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/HomeFragment.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    private ViewPager mViewPagerHome;
+
+    public HomeFragment() {
+        //No-op
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        initData();
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/fragment/HomeFragment.java]: when you declare data or handle somethings pls add bellow the init view is better

File: app/src/at-nhatnguyen/res/layout/item_home_viewpager.xml [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+            <TextView
+                android:id="@+id/tvVietNam"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:gravity="center"
+                android:textSize="20dp" />
+        </LinearLayout>
+    </android.support.v7.widget.CardView>
+</LinearLayout>
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/res/layout/item_home_viewpager.xml]: break line

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/MainActivity.java [hunk 12]
```diff
@@ -10,10 +10,12 @@
 import vn.asiantech.internship.ui.calculator.CalculatorActivity;
 import vn.asiantech.internship.ui.imageview.ImageActivity;
 import vn.asiantech.internship.ui.login.LoginActivity;
+import vn.asiantech.internship.ui.login.LoginActivity;
```
Comment (at-dev-support-01) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/MainActivity.java]: WARNING: [DuplicateImports] Avoid duplicate imports such as 'vn.asiantech.internship.ui.login.LoginActivity'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#DuplicateImports

**Comment by at-nhatnguyen** [comment]
@at-quocnguyen please review my task, thank you

File: app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+    public static UpdateDialog newInstance() {
+        Bundle args = new Bundle();
+        UpdateDialog fragment = new UpdateDialog();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        getDialog().setTitle("Update Information");
```
Comment (at-quocnguyen) [inline app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java]: No need set title dialog fragment please remove

Comment (at-quocnguyen) [reply app/src/at-nhatnguyen/java/vn/asiantech/internship/ui/viewpager/UpdateDialog.java]: if you wanna setting please move to onCreateDialog

**Comment by at-nhatnguyen** [comment]
@at-quocnguyen please review my task, thank you

**Pull request merged**